lib/IMAP/FolderMapper.php

```diff
@@ -62,12 +62,19 @@ public function getFolders(Account $account, Horde_Imap_Client_Socket $client,
 			'special_use' => true,
 		]);
 
-		return array_filter(array_map(function (array $mailbox) use ($account) {
+		return array_filter(array_map(function (array $mailbox) use ($account, $client) {
 			if (in_array($mailbox['mailbox']->utf8, self::DOVECOT_SIEVE_FOLDERS, true)) {
 				// This is a special folder that must not be shown
 				return null;
 			}
 
+			// ignore folders which cause errors on access (i.e. server-side SYSTEM I/O ERROR)
+			try {
+				$client->status($mailbox["mailbox"]);
+			} catch (Horde_Imap_Client_Exception $e) {
+				return null;
```

Comment:
so this will also effectively make any IMAP hiccups hide the mailbox until the next mailbox sync

can we catch the specific error code? see the constants on ``Horde_Imap_Client_Exception``, we can compare those to $e->getCode().

Reply:
Thanks Christoph, I completely agree...just wasn't sure how to get the error code properly (https://dev.horde.org/api/master/lib/Imap_Client/classes/Horde_Imap_Client_Exception.html only mentions the setter-Method). Let me check and update the request.

Reply:
>  (https://dev.horde.org/api/master/lib/Imap_Client/classes/Horde_Imap_Client_Exception.html only mentions the setter-Method)

I think that's because the `getCode` isn't overwritten, they just pass the code down into php's `\Exception` base class

Reply:
The change is in (conveniently enough the relevant error in our case is UNSPECIFIED).

Reply:
Well, I guess that is OK. The more common errors like timeout or authentication will have a distinct error code.